### ISSUE TITLE
Updating Az module version 11.3.0 using Update-Module command fails

### ISSUE THREAD

**Issue by YuKogasaka** [Azure/azure-powershell#24112]
### Description

An attempt to update the Az module to version 11.3.0 using the Update-Module command failed with the following error message:

```
Install-Package: The version '11.3.0' of the module 'Az' being installed is not catalog signed. Ensure that the version '11.3.0' of the
module 'Az' has the catalog file 'Az.cat' and signed with the same publisher 'CN=Microsoft Corporation, O=Microsoft
Corporation, L=Redmond, S=Washington, C=US' as the previously-installed module 'Az' with version '11.2.0' under the
directory 'C:\Program Files\PowerShell\Modules\Az\11.2.0'. If you still want to install or update, use
-SkipPublisherCheck parameter.
```

Using the Install-Module command also fails with a similar error.

Note : Update-Module did not have a -SkipPublisherCheck parameter.

### Issue script & Debug output

```PowerShell
PS C:\> $DebugPreference='Continue'
PS C:\> Update-Module Az -Scope AllUsers
DEBUG: 00:00:00.0000091 Calling New() : MethodName = 'GetDynamicOptions'
DEBUG: 00:00:00.0000478 WarningAction: SilentlyContinue
DEBUG: 00:00:00.0000690 Name: Az
DEBUG: 00:00:00.0000882 ErrorAction: SilentlyContinue
DEBUG: 00:00:00.0030128 INVOKING PowerShell Fn Get-DynamicOptions with args Provider that has length 1
DEBUG: 00:00:00.0068873 In PowerShellGet Provider - 'Get-DynamicOptions'.
DEBUG: 00:00:00.0115024 Done calling powershell «Get-DynamicOptions» «PSModule»
DEBUG: 00:00:00.0181882 Calling New() : MethodName = 'GetDynamicOptions'
DEBUG: 00:00:00.0182392 WarningAction: SilentlyContinue
DEBUG: 00:00:00.0182705 Name: Az
DEBUG: 00:00:00.0182990 ErrorAction: SilentlyContinue
DEBUG: 00:00:00.0206759 INVOKING PowerShell Fn Get-DynamicOptions with args Install that has length 1
DEBUG: 00:00:00.0236986 In PowerShellGet Provider - 'Get-DynamicOptions'.
DEBUG: 00:00:00.0286729 Done calling powershell «Get-DynamicOptions» «PSModule»
DEBUG: 00:00:00.0394732 Calling New() : MethodName = 'GetInstalledPackages'
DEBUG: 00:00:00.0404712 ErrorAction: SilentlyContinue
DEBUG: 00:00:00.0414962 WarningAction: SilentlyContinue
DEBUG: 00:00:00.0424867 MessageResolver: Microsoft.PowerShell.PackageManagement.Cmdlets.GetMessageString
DEBUG: 00:00:00.0434899 ProviderName: PowerShellGet
DEBUG: 00:00:00.0444675 Type: Module
DEBUG: 00:00:00.0455100 Name: Az
DEBUG: 00:00:00.0518317 INVOKING PowerShell Fn Get-InstalledPackage with args Az, , ,  that has length 4
DEBUG: 00:00:00.0561051 In PowerShellGet Provider - 'Get-InstalledPackage'.
DEBUG: 00:00:00.0569039 OPTION: ErrorAction => SilentlyContinue
DEBUG: 00:00:00.0793043 OPTION: WarningAction => SilentlyContinue
DEBUG: 00:00:00.0799429 OPTION: MessageResolver => Microsoft.PowerShell.PackageManagement.Cmdlets.GetMessageString
DEBUG: 00:00:00.0804995 OPTION: ProviderName => PowerShellGet
DEBUG: 00:00:00.0809537 OPTION: Type => Module
DEBUG: 00:00:00.0816395 OPTION: Name => Az
DEBUG: 00:00:02.9456666 Done calling powershell «Get-InstalledPackage» «PSModule»
DEBUG: 00:00:00.0000039 Calling New() : MethodName = 'ResolvePackageSources'
DEBUG: 00:00:00.0000336 Name: Az
DEBUG: 00:00:00.0000417 Source: PSGallery
DEBUG: 00:00:00.0015557 INVOKING PowerShell Fn Resolve-PackageSource with args  that has length 0
DEBUG: 00:00:00.0031955 In PowerShellGet Provider - 'Resolve-PackageSource'.
DEBUG: 00:00:00.0039038 Yielding package source for PSGallery at location https://www.powershellgallery.com/api/v2
DEBUG: 00:00:00.0058313 Done calling powershell «Resolve-PackageSource» «PSModule»
DEBUG: 00:00:00.0088689 Calling New() : MethodName = 'GetDynamicOptions'
DEBUG: 00:00:00.0088837 Name: Az
DEBUG: 00:00:00.0088911 Source: PSGallery
DEBUG: 00:00:00.0100718 INVOKING PowerShell Fn Get-DynamicOptions with args Provider that has length 1
DEBUG: 00:00:00.0116732 In PowerShellGet Provider - 'Get-DynamicOptions'.
DEBUG: 00:00:00.0136941 Done calling powershell «Get-DynamicOptions» «PSModule»
DEBUG: 00:00:00.0196225 Calling New() : MethodName = 'GetDynamicOptions'
DEBUG: 00:00:00.0196386 Name: Az
DEBUG: 00:00:00.0196461 Source: PSGallery
DEBUG: 00:00:00.0208532 INVOKING PowerShell Fn Get-DynamicOptions with args Source that has length 1
DEBUG: 00:00:00.0225054 In PowerShellGet Provider - 'Get-DynamicOptions'.
DEBUG: 00:00:00.0246608 Done calling powershell «Get-DynamicOptions» «PSModule»
DEBUG: 00:00:00.0276784 Calling New() : MethodName = 'GetDynamicOptions'
DEBUG: 00:00:00.0276920 Name: Az
DEBUG: 00:00:00.0276992 Source: PSGallery
DEBUG: 00:00:00.0291059 INVOKING PowerShell Fn Get-DynamicOptions with args Package that has length 1
DEBUG: 00:00:00.0316653 In PowerShellGet Provider - 'Get-DynamicOptions'.
DEBUG: 00:00:00.0341951 Done calling powershell «Get-DynamicOptions» «PSModule»
DEBUG: 00:00:00.0373498 Calling New() : MethodName = 'GetDynamicOptions'
DEBUG: 00:00:00.0373631 Name: Az
DEBUG: 00:00:00.0373704 Source: PSGallery
DEBUG: 00:00:00.0386710 INVOKING PowerShell Fn Get-DynamicOptions with args Install that has length 1
DEBUG: 00:00:00.0404851 In PowerShellGet Provider - 'Get-DynamicOptions'.
DEBUG: 00:00:00.0428246 Done calling powershell «Get-DynamicOptions» «PSModule»
DEBUG: 00:00:00.0592012 Calling New() : MethodName = 'ResolvePackageSources'
DEBUG: 00:00:00.0595568 InstallUpdate: True
DEBUG: 00:00:00.0599263 Source: PSGallery
DEBUG: 00:00:00.0624891 MessageResolver: Microsoft.PowerShell.PackageManagement.Cmdlets.GetMessageString
DEBUG: 00:00:00.0631244 Type: Module
DEBUG: 00:00:00.0674069 PackageManagementProvider: NuGet
DEBUG: 00:00:00.0678686 Scope: AllUsers
DEBUG: 00:00:00.0681872 ProviderName: PowerShellGet
DEBUG: 00:00:00.0685116 Name: Az
DEBUG: 00:00:00.0702840 INVOKING PowerShell Fn Resolve-PackageSource with args  that has length 0
DEBUG: 00:00:00.0719596 In PowerShellGet Provider - 'Resolve-PackageSource'.
DEBUG: 00:00:00.0729061 Yielding package source for PSGallery at location https://www.powershellgallery.com/api/v2
DEBUG: 00:00:00.0750306 Done calling powershell «Resolve-PackageSource» «PSModule»
DEBUG: 00:00:00.0752037 Calling SearchForPackages. Name='Az'
DEBUG: 00:00:00.0760018 PackageProvider::FindPackage with name Az
DEBUG: 00:00:00.0779446 Calling SearchForPackages After Select 1
DEBUG: 00:00:00.0794094 Calling New() : MethodName = 'FindPackage'
DEBUG: 00:00:00.0797481 InstallUpdate: True
DEBUG: 00:00:00.0801133 Source: PSGallery
DEBUG: 00:00:00.0805713 MessageResolver: Microsoft.PowerShell.PackageManagement.Cmdlets.GetMessageString
DEBUG: 00:00:00.0808756 Type: Module
DEBUG: 00:00:00.0811026 PackageManagementProvider: NuGet
DEBUG: 00:00:00.0813482 Scope: AllUsers
DEBUG: 00:00:00.0815724 ProviderName: PowerShellGet
DEBUG: 00:00:00.0818310 Name: Az
DEBUG: 00:00:00.0835360 INVOKING PowerShell Fn Find-Package with args System.String[], , ,  that has length 4
DEBUG: 00:00:00.0853840 In PowerShellGet Provider - 'Find-Package'.
DEBUG: 00:00:00.0856341 OPTION: InstallUpdate => True
DEBUG: 00:00:00.0859049 OPTION: Source => PSGallery
DEBUG: 00:00:00.0862089 OPTION: MessageResolver => Microsoft.PowerShell.PackageManagement.Cmdlets.GetMessageString
DEBUG: 00:00:00.0865794 OPTION: Type => Module
DEBUG: 00:00:00.0868114 OPTION: PackageManagementProvider => NuGet
DEBUG: 00:00:00.0872782 OPTION: Scope => AllUsers
DEBUG: 00:00:00.0875981 OPTION: ProviderName => PowerShellGet
DEBUG: 00:00:00.0880120 OPTION: Name => Az
DEBUG: 00:00:00.0903459 PackageProvider::FindPackage with name Az
DEBUG: 00:00:00.0938644 Calling 'NuGet'::'FindPackage' - name='Az', requiredVersion='',minimumVersion='', maximumVersion='''.
DEBUG: 00:00:00.0943648 Iterating 'Az'.
DEBUG: 00:00:00.0946883 There are '1' registered sources in 'NuGet' provider.
DEBUG: 00:00:00.0950125 Source 'https://www.powershellgallery.com/api/v2' is not one of the registered sources in 'NuGet' provider.
DEBUG: 00:00:00.0953747 Source 'https://www.powershellgallery.com/api/v2' is validated.
DEBUG: 00:00:00.0956536 Calling 'NuGetRequest'::'GetPackageById', 'Az'.
DEBUG: 00:00:00.0959015 Calling 'NuGetPackageRepository'::'FindPackagesById', 'Az'.
DEBUG: 00:00:00.0962796 Calling 'NuGetPackageFeed2'::'FindPackage', 'Az'.
DEBUG: 00:00:00.0965982 Calling 'NuGetClient'::'FindPackage'.
DEBUG: 00:00:00.0971074 Downloading 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='Az'&$skip=0&$top=40'.
DEBUG: 00:00:00.3342052 Completed downloading 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='Az'&$skip=0&$top=40'.
DEBUG: 00:00:00.3370283 Downloading 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='Az'&$skip=0&$top=40'.
DEBUG: 00:00:00.5267679 Completed downloading 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='Az'&$skip=0&$top=40'.
DEBUG: 00:00:00.5420545 Downloading 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='Az'&$skip=80&$top=40'.
DEBUG: 00:00:00.5420545 Downloading 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='Az'&$skip=40&$top=40'.
DEBUG: 00:00:00.5436453 Downloading 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='Az'&$skip=120&$top=40'.
DEBUG: 00:00:00.5455593 Downloading 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='Az'&$skip=160&$top=40'.
DEBUG: 00:00:00.7413273 Completed downloading 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='Az'&$skip=80&$top=40'.
DEBUG: 00:00:00.7437609 '21' packages received in the last request.
DEBUG: 00:00:00.7503329 Completed downloading 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='Az'&$skip=160&$top=40'.
DEBUG: 00:00:00.7516434 '0' packages received in the last request.
DEBUG: 00:00:00.7814685 Completed downloading 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='Az'&$skip=120&$top=40'.
DEBUG: 00:00:00.7833569 '0' packages received in the last request.
DEBUG: 00:00:00.9563321 Completed downloading 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='Az'&$skip=40&$top=40'.
DEBUG: 00:00:00.9840024 Completed iterating for 'Az'.
DEBUG: 00:00:01.0162740 Done calling powershell «Find-Package» «PSModule»
DEBUG: 00:00:01.0260325 Calling New() : MethodName = 'GetInstalledPackages'
DEBUG: 00:00:01.0271550 InstallUpdate: True
DEBUG: 00:00:01.0283880 Source: PSGallery
DEBUG: 00:00:01.0291011 MessageResolver: Microsoft.PowerShell.PackageManagement.Cmdlets.GetMessageString
DEBUG: 00:00:01.0298538 Type: Module
DEBUG: 00:00:01.0304412 PackageManagementProvider: NuGet
DEBUG: 00:00:01.0310835 Scope: AllUsers
DEBUG: 00:00:01.0324206 ProviderName: PowerShellGet
DEBUG: 00:00:01.0329376 Name: Az
DEBUG: 00:00:01.0344761 INVOKING PowerShell Fn Get-InstalledPackage with args Az, 11.3.0, ,  that has length 4
DEBUG: 00:00:01.0366155 In PowerShellGet Provider - 'Get-InstalledPackage'.
DEBUG: 00:00:01.0371435 OPTION: InstallUpdate => True
DEBUG: 00:00:01.0380326 OPTION: Source => PSGallery
DEBUG: 00:00:01.0388782 OPTION: MessageResolver => Microsoft.PowerShell.PackageManagement.Cmdlets.GetMessageString
DEBUG: 00:00:01.0392157 OPTION: Type => Module
DEBUG: 00:00:01.0399187 OPTION: PackageManagementProvider => NuGet
DEBUG: 00:00:01.0403671 OPTION: Scope => AllUsers
DEBUG: 00:00:01.0409602 OPTION: ProviderName => PowerShellGet
DEBUG: 00:00:01.0413484 OPTION: Name => Az
DEBUG: 00:00:04.0207849 PowerShell Script 'PSModule' Function 'Get-InstalledPackage' returns null.
DEBUG: 00:00:04.0230543 Done calling powershell «Get-InstalledPackage» «PSModule»

Untrusted repository
You are installing the modules from an untrusted repository. If you trust this repository, change its
InstallationPolicy value by running the Set-PSRepository cmdlet. Are you sure you want to install the modules from
'PSGallery'?
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "N"): A
DEBUG: 00:00:07.1493952 Calling New() : MethodName = 'InstallPackage'
DEBUG: 00:00:07.1501883 InstallUpdate: True
DEBUG: 00:00:07.1508922 Source: PSGallery
DEBUG: 00:00:07.1517326 MessageResolver: Microsoft.PowerShell.PackageManagement.Cmdlets.GetMessageString
DEBUG: 00:00:07.1523128 Type: Module
DEBUG: 00:00:07.1528143 PackageManagementProvider: NuGet
DEBUG: 00:00:07.1531387 Scope: AllUsers
DEBUG: 00:00:07.1542102 ProviderName: PowerShellGet
DEBUG: 00:00:07.1544198 Name: Az
DEBUG: 00:00:07.1562103 INVOKING PowerShell Fn Install-Package with args NuGet|Az|11.3.0|https://www.powershellgallery.com/api/v2|Module that has length 1
DEBUG: 00:00:07.1587224 In PowerShellGet Provider - 'Install-Package'.
DEBUG: 00:00:07.1702723 In PowerShellGet Provider - 'Install-PackageUtility'.
DEBUG: 00:00:07.1712608 The FastPackageReference is 'NuGet|Az|11.3.0|https://www.powershellgallery.com/api/v2|Module'.
DEBUG: 00:00:07.1729663 OPTION: InstallUpdate => True
DEBUG: 00:00:07.1737499 OPTION: Source => PSGallery
DEBUG: 00:00:07.1743316 OPTION: MessageResolver => Microsoft.PowerShell.PackageManagement.Cmdlets.GetMessageString
DEBUG: 00:00:07.1750986 OPTION: Type => Module
DEBUG: 00:00:07.1758320 OPTION: PackageManagementProvider => NuGet
DEBUG: 00:00:07.1765917 OPTION: Scope => AllUsers
DEBUG: 00:00:07.1774793 OPTION: ProviderName => PowerShellGet
DEBUG: 00:00:07.1782444 OPTION: Name => Az
DEBUG: 00:00:07.1805994 ArtifactType is Module
DEBUG: 00:00:13.5823562 Calling 'NuGet'::'InstallPackage', '$aHR0cHM6Ly93d3cucG93ZXJzaGVsbGdhbGxlcnkuY29tL2FwaS92Mg==\QXo=\MTEuMy4w\aHR0cHM6Ly93d3cucG93ZXJzaGVsbGdhbGxlcnkuY29tL2FwaS92Mg==\cG93ZXJzaGVsbGdldA=='.
DEBUG: 00:00:13.5840435 Calling 'NuGetRequest'::'GetPackageByFastpath', '$aHR0cHM6Ly93d3cucG93ZXJzaGVsbGdhbGxlcnkuY29tL2FwaS92Mg==\QXo=\MTEuMy4w\aHR0cHM6Ly93d3cucG93ZXJzaGVsbGdhbGxlcnkuY29tL2FwaS92Mg==\cG93ZXJzaGVsbGdldA=='.
DEBUG: 00:00:13.5854800 Calling 'NuGetRequest'::'ResolvePackageSource', 'https://www.powershellgallery.com/api/v2'.
DEBUG: 00:00:13.5869969 Calling 'NuGetRequest'::'FindRegisteredSource', 'https://www.powershellgallery.com/api/v2'.
DEBUG: 00:00:13.5882713 Source 'https://www.powershellgallery.com/api/v2' is not one of the registered sources in 'NuGet' provider.
DEBUG: 00:00:13.7962218 Calling 'NuGetPackageRepository'::'FindPackage', 'Az'.
DEBUG: 00:00:13.7979811 Calling 'NuGetPackageFeed2'::'FindPackage', 'Az'.
DEBUG: 00:00:13.7994795 Calling 'NuGetClient'::'FindPackage'.
DEBUG: 00:00:13.8020376 Downloading 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='Az'&$skip=0&$top=40'.
DEBUG: 00:00:14.0439537 Completed downloading 'https://www.powershellgallery.com/api/v2/FindPackagesById()?id='Az'&$skip=0&$top=40'.
DEBUG: 00:00:14.0807559 'Package version' is '11.3.0'.
DEBUG: 00:00:14.0820182 'Request's Destination' is 'C:\Users\kogasaka\AppData\Local\Temp\1477486535'.
DEBUG: 00:00:14.0828795 Calling 'NuGetPackageRepository'::'InstallPackage'.
DEBUG: 00:00:14.0842241 Calling 'NuGetFilesFeed2'::'InstallPackage'.
DEBUG: 00:00:14.0859651 Calling 'NuGetFilesFeed2'::'InstallPackage', '$aHR0cHM6Ly93d3cucG93ZXJzaGVsbGdhbGxlcnkuY29tL2FwaS92Mg==\QXo=\MTEuMy4w\aHR0cHM6Ly93d3cucG93ZXJzaGVsbGdhbGxlcnkuY29tL2FwaS92Mg==\cG93ZXJzaGVsbGdldA=='.
DEBUG: 00:00:14.0873998 Calling 'NuGetClient'::'GetPackageDependencies'.
DEBUG: 00:00:14.1280986 Returning the call 'NuGetClient'::'GetPackageDependencies'.
DEBUG: 00:00:14.1316465 Calling 'NuGetClient'::'InstallPackage'.
DEBUG: 00:00:14.4468437 Calling 'NuGetClient'::'InstallPackageLocal'.
DEBUG: 00:00:14.4537756 C:\Users\kogasaka\AppData\Local\Temp\1477486535\Az.11.3.0\Az.11.3.0.nupkg
DEBUG: 00:00:14.5208197 Returning the call 'NuGetClient'::'InstallPackageLocal'.
DEBUG: 00:00:14.5304705 Returning the call 'NuGetClient'::'InstallSinglePackage'.
DEBUG: 00:00:14.5313251 Returning the call 'NuGetFilesFeed2'::'InstallPackage'.
DEBUG: 00:00:17.9773655 Catalog file 'Az.cat' is not found in the contents of the previously-installed module 'Az' with the same name.
DEBUG: 00:00:17.9779628 Using the previously-installed module 'Az' with version '11.2.0' under 'C:\Program Files\PowerShell\Modules\Az\11.2.0' for getting the publisher details.
DEBUG: 00:00:17.9786598 Using the 'C:\Program Files\PowerShell\Modules\Az\11.2.0\Az.psd1' file for getting the authenticode signature.
DEBUG: 00:00:18.0843918 For publisher validation, using the previously-installed module 'Az' with version '11.2.0' under 'C:\Program Files\PowerShell\Modules\Az\11.2.0' with publisher name 'CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US' from root certificate authority 'CN=Microsoft Root Certificate Authority 2011, O=Microsoft Corporation, L=Redmond, S=Washington, C=US'. Is this module signed by Microsoft: 'True'.
DEBUG: 00:00:18.0855115 Using the 'Az.psd1' file for getting the authenticode signature.
DEBUG: 00:00:18.0888735 For publisher validation, current module 'Az' with version '11.3.0' with publisher name '' from root certificate authority ''. Is this module signed by Microsoft: 'False'.
DEBUG: 00:00:18.0897897 Previously-installed module publisher: CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US
DEBUG: 00:00:18.0900524 Current module publisher:
DEBUG: 00:00:18.0906230 Is previously-installed module signed by Microsoft: True
DEBUG: 00:00:18.0912903 Is current module signed by Microsoft: False
Install-Package: The version '11.3.0' of the module 'Az' being installed is not catalog signed. Ensure that the version '11.3.0' of the
module 'Az' has the catalog file 'Az.cat' and signed with the same publisher 'CN=Microsoft Corporation, O=Microsoft
Corporation, L=Redmond, S=Washington, C=US' as the previously-installed module 'Az' with version '11.2.0' under the
directory 'C:\Program Files\PowerShell\Modules\Az\11.2.0'. If you still want to install or update, use
-SkipPublisherCheck parameter.
```

### Environment data

```PowerShell
PS C:\> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.4.1
PSEdition                      Core
GitCommitId                    7.4.1
OS                             Microsoft Windows 10.0.22631
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Module versions

```PowerShell
PS C:\> Get-Module Az*

ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Script     11.2.0                Az
Script     2.15.1                Az.Accounts                         {Add-AzEnvironment, Clear-AzConfig, Clear-AzConte…
Script     2.0.0                 Az.Advisor                          {Disable-AzAdvisorRecommendation, Enable-AzAdviso…
Script     6.0.0                 Az.Aks                              {Disable-AzAksAddOn, Enable-AzAksAddOn, Get-AzAks…
Script     1.1.4                 Az.AnalysisServices                 {Add-AzAnalysisServicesAccount, Export-AzAnalysis…
Script     4.0.2                 Az.ApiManagement                    {Add-AzApiManagementApiToGateway, Add-AzApiManage…
Script     1.0.0                 Az.App                              {Disable-AzContainerAppRevision, Enable-AzContain…
Script     1.3.0                 Az.AppConfiguration                 {Clear-AzAppConfigurationDeletedStore, Get-AzAppC…
Script     2.2.3                 Az.ApplicationInsights              {Get-AzApplicationInsights, Get-AzApplicationInsi…
Script     1.0.0                 Az.ArcResourceBridge                {Get-AzArcResourceBridge, Get-AzArcResourceBridge…
Script     2.0.0                 Az.Attestation                      {Add-AzAttestationPolicySigner, Get-AzAttestation…
Script     1.0.0                 Az.Automanage                       {Get-AzAutomanageBestPractice, Get-AzAutomanageCo…
Script     1.10.0                Az.Automation                       {Export-AzAutomationDscConfiguration, Export-AzAu…
Script     3.5.0                 Az.Batch                            {Disable-AzBatchAutoScale, Disable-AzBatchCompute…
Script     2.0.3                 Az.Billing                          {Get-AzBillingAccount, Get-AzBillingInvoice, Get-…
Script     3.1.1                 Az.Cdn                              {Clear-AzCdnEndpointContent, Clear-AzFrontDoorCdn…
Script     2.0.0                 Az.CloudService                     {Get-AzCloudService, Get-AzCloudServiceInstanceVi…
Script     1.14.1                Az.CognitiveServices                {Add-AzCognitiveServicesAccountNetworkRule, Get-A…
Script     7.1.1                 Az.Compute                          {Add-AzImageDataDisk, Add-AzVhd, Add-AzVMAddition…
Script     1.0.0                 Az.ConfidentialLedger               {Get-AzConfidentialLedger, New-AzConfidentialLedg…
Script     4.0.0                 Az.ContainerInstance                {Add-AzContainerInstanceOutput, Get-AzContainerGr…
Script     4.1.3                 Az.ContainerRegistry                {Connect-AzContainerRegistry, Get-AzContainerRegi…
Script     1.14.0                Az.CosmosDB                         {Get-AzCosmosDBAccount, Get-AzCosmosDBAccountKey,…
Script     1.1.0                 Az.DataBoxEdge                      {Get-AzDataBoxEdgeBandwidthSchedule, Get-AzDataBo…
Script     1.7.1                 Az.Databricks                       {Get-AzDatabricksAccessConnector, Get-AzDatabrick…
Script     1.18.0                Az.DataFactory                      {Add-AzDataFactoryV2DataFlowDebugSessionPackage, …
Script     1.0.3                 Az.DataLakeAnalytics                {Add-AzDataLakeAnalyticsDataSource, Add-AzDataLak…
Script     1.3.0                 Az.DataLakeStore                    {Add-AzDataLakeStoreFirewallRule, Add-AzDataLakeS…
Script     2.2.0                 Az.DataProtection                   {Backup-AzDataProtectionBackupInstanceAdhoc, Edit…
Script     1.0.1                 Az.DataShare                        {Get-AzDataShare, Get-AzDataShareAccount, Get-AzD…
Script     1.1.0                 Az.DeploymentManager                {Get-AzDeploymentManagerArtifactSource, Get-AzDep…
Script     4.3.0                 Az.DesktopVirtualization            {Disconnect-AzWvdUserSession, Expand-AzWvdMsixIma…
Script     1.1.0                 Az.DevCenter                        {Deploy-AzDevCenterUserEnvironment, Get-AzDevCent…
Script     1.0.2                 Az.DevTestLabs                      {Get-AzDtlAllowedVMSizesPolicy, Get-AzDtlAutoShut…
Script     1.2.0                 Az.Dns                              {Add-AzDnsRecordConfig, Get-AzDnsRecordSet, Get-A…
Script     1.6.0                 Az.EventGrid                        {Enable-AzEventGridPartnerTopic, Get-AzEventGridC…
Script     4.2.0                 Az.EventHub                         {New-AzEventHubAuthorizationRuleSASToken, Approve…
Script     1.10.0                Az.FrontDoor                        {Disable-AzFrontDoorCustomDomainHttps, Enable-AzF…
Script     4.0.7                 Az.Functions                        {Get-AzFunctionApp, Get-AzFunctionAppAvailableLoc…
Script     6.1.0                 Az.HDInsight                        {Add-AzHDInsightClusterIdentity, Add-AzHDInsightC…
Script     2.0.0                 Az.HealthcareApis                   {Get-AzHealthcareApisService, Get-AzHealthcareApi…
Script     2.7.5                 Az.IotHub                           {Add-AzIotHubCertificate, Add-AzIotHubConfigurati…
Script     5.1.0                 Az.KeyVault                         {Add-AzKeyVaultCertificate, Add-AzKeyVaultCertifi…
Script     2.3.0                 Az.Kusto                            {Add-AzKustoClusterLanguageExtension, Add-AzKusto…
Script     1.0.0                 Az.LoadTesting                      {Get-AzLoad, New-AzLoad, Remove-AzLoad, Update-Az…
Script     1.5.0                 Az.LogicApp                         {Get-AzIntegrationAccount, Get-AzIntegrationAccou…
Script     1.1.3                 Az.MachineLearning                  {Add-AzMlWebServiceRegionalProperty, Export-AzMlW…
Script     1.0.0                 Az.MachineLearningServices          {Get-AzMLServiceQuota, Get-AzMLServiceUsage, Get-…
Script     1.4.1                 Az.Maintenance                      {Get-AzApplyUpdate, Get-AzConfigurationAssignment…
Script     1.2.0                 Az.ManagedServiceIdentity           {Get-AzFederatedIdentityCredential, Get-AzSystemA…
Script     3.0.0                 Az.ManagedServices                  {Get-AzManagedServicesAssignment, Get-AzManagedSe…
Script     2.0.0                 Az.MarketplaceOrdering              {Get-AzMarketplaceTerms, Invoke-AzMarketplaceSign…
Script     1.1.2                 Az.Media                            {Get-AzMediaService, Get-AzMediaServiceKey, Get-A…
Script     2.2.0                 Az.Migrate                          {Get-AzMigrateDiscoveredServer, Get-AzMigrateJob,…
Script     5.0.0                 Az.Monitor                          {Add-AzLogProfile, Add-AzMetricAlertRule, Add-AzM…
Script     1.1.1                 Az.MySql                            {Get-AzMySqlConfiguration, Get-AzMySqlConnectionS…
Script     7.3.0                 Az.Network                          {Add-AzApplicationGatewayAuthenticationCertificat…
Script     1.0.0                 Az.NetworkCloud                     {Deploy-AzNetworkCloudCluster, Disable-AzNetworkC…
Script     1.1.2                 Az.NotificationHubs                 {Get-AzNotificationHub, Get-AzNotificationHubAuth…
Script     3.2.0                 Az.OperationalInsights              {Disable-AzOperationalInsightsIISLogCollection, D…
Script     1.6.4                 Az.PolicyInsights                   {Get-AzPolicyAttestation, Get-AzPolicyEvent, Get-…
Script     1.1.0                 Az.PostgreSql                       {Get-AzPostgreSqlConfiguration, Get-AzPostgreSqlC…
Script     2.0.0                 Az.PowerBIEmbedded                  {Get-AzPowerBIEmbeddedCapacity, New-AzPowerBIEmbe…
Script     1.0.4                 Az.PrivateDns                       {Add-AzPrivateDnsRecordConfig, Get-AzPrivateDnsRe…
Script     6.7.0                 Az.RecoveryServices                 {Add-AzRecoveryServicesAsrReplicationProtectedIte…
Script     1.8.1                 Az.RedisCache                       {Export-AzRedisCache, Get-AzRedisCache, Get-AzRed…
Script     1.2.0                 Az.RedisEnterpriseCache             {Export-AzRedisEnterpriseCache, Get-AzRedisEnterp…
Script     2.0.0                 Az.Relay                            {Get-AzRelayAuthorizationRule, Get-AzRelayHybridC…
Script     1.2.0                 Az.ResourceMover                    {Add-AzResourceMoverMoveResource, Get-AzResourceM…
Script     6.13.0                Az.Resources                        {Export-AzResourceGroup, Export-AzTemplateSpec, G…
Script     1.5.1                 Az.Security                         {Add-AzSecurityAdaptiveNetworkHardening, Add-AzSe…
Script     3.1.1                 Az.SecurityInsights                 {Get-AzSentinelAlertRule, Get-AzSentinelAlertRule…
Script     3.0.0                 Az.ServiceBus                       {New-AzServiceBusAuthorizationRuleSASToken, Test-…
Script     3.3.2                 Az.ServiceFabric                    {Add-AzServiceFabricClientCertificate, Add-AzServ…
Script     2.0.0                 Az.SignalR                          {Get-AzSignalR, Get-AzSignalRKey, Get-AzSignalRUs…
Script     4.13.0                Az.Sql                              {Add-AzSqlDatabaseToFailoverGroup, Add-AzSqlElast…
Script     2.1.0                 Az.SqlVirtualMachine                {Get-AzAvailabilityGroupListener, Get-AzSqlVM, Ge…
Script     2.2.3                 Az.StackHCI                         {Add-AzStackHCIVMAttestation, Disable-AzStackHCIA…
Script     6.1.0                 Az.Storage                          {Add-AzRmStorageContainerLegalHold, Add-AzStorage…
Script     1.3.0                 Az.StorageMover                     {Get-AzStorageMover, Get-AzStorageMoverAgent, Get…
Script     2.1.1                 Az.StorageSync                      {Get-AzStorageSyncCloudEndpoint, Get-AzStorageSyn…
Script     2.0.0                 Az.StreamAnalytics                  {Get-AzStreamAnalyticsCluster, Get-AzStreamAnalyt…
Script     1.0.0                 Az.Support                          {Get-AzSupportProblemClassification, Get-AzSuppor…
Script     3.0.5                 Az.Synapse                          {Add-AzSynapseDataFlowDebugSessionPackage, Add-Az…
Script     1.2.2                 Az.TrafficManager                   {Add-AzTrafficManagerCustomHeaderToEndpoint, Add-…
Script     3.1.2                 Az.Websites                         {Add-AzWebAppAccessRestrictionRule, Add-AzWebAppT…
```

**Comment by isra-fel** [comment]
Thanks for reporting!
The signing of Az.psd1 was skipped for some reason in v11.3.0 . We are investigating.
As a temporary work-around, please uninstall the previous version then install Az again.

```powershell
Uninstall-Module Az
Install-Module Az -RequiredVersion 11.3.0 -Repository PSGallery
```

**Comment by c4rlosmarin** [comment]
> Thanks for reporting! The signing of Az.psd1 was skipped for some reason in v11.3.0 . We are investigating. As a temporary work-around, please uninstall the previous version then install Az again.
> 
> ```powershell
> Uninstall-Module Az -AllVersions
> Install-Module Az -RequiredVersion 11.3.0 -Repository PSGallery
> ```

I just tried this and got the same error, fortunately I was able to reinstall version 11.2.0:

![image](https://github.com/Azure/azure-powershell/assets/17365297/7d87494a-164d-4f97-aac7-c61f69b2e80a)

**Comment by c4rlosmarin** [comment]
> `Install-Module -Name Az -Repository PSGallery -Force -SkipPublisherCheck`
> 
> This works for me.

Confirmed, thanks.

**Comment by wyunchi-ms** [comment]
This issue is fixed in Az 11.3.1.
